### DISCUSSION TITLE
dnsdist 1.8: Force to listen only TCP

### DISCUSSION THREAD

**Discussion by dtouzeau** [PowerDNS/pdns#13171] [General]
Clients usually switch to tcp queries if UDP is not available.
But we did not find any information to force dnsdist to not listen UDP protocol.
By default it listens UDP/TCP, 
how to remove UDP ?

**Comment by Habbie** [comment]
Hello! We recently opened a Discussions forum on GitHub, so we can handle conversations that are not bug reports or feature requests - such as yours. Your ticket is the first one we converted, so please bear with us while we figure out this process.

**Comment by Habbie** [comment]
> Clients usually switch to tcp queries if UDP is not available.

I have to ask: do they? I've not seen much documentation about clients doing this.

> how to remove UDP ?

This is currently not possible.

A question from me: why do you want this? What problem are you trying to solve?

**Reply by rgacogne** [reply]
> Clients usually switch to tcp queries if UDP is not available.

PowerDNS Recursor won't, as an example.

**Comment by franklouwers** [comment]
If you absolutely do not want resolving over UDP, you could set a `DNSAction.Truncate`, which will force the udp client to switch to TCP. It will still listen on udp and send 1 udp packet to the client.

But let's take a step back here. What problem are you trying to solve? Because for general resolvers/auth-servers, blocking UDP will not work at all, unless you'd go the DoH/DoT-only route, which is something dnsdist is certainly capable of.

**Comment by franklouwers** [comment]
propose to close/resolve this, no reply after 3 weeks

**Comment by rgacogne** [comment]
I'm going to close this discussion since it seems stale, but I'll of course re-open it if needed.
